app/models/delete_queue_job.rb

```diff
@@ -99,12 +102,8 @@ def perform
         object.study_file_bundle.destroy
       end
 
-      # queue study file object for deletion, set file_type to DELETE to prevent it from being picked up in any queries
-      # use assign_attributes and save!(validate: false) to avoid any validation issues that might prevent saving
-      # this ensures file is queued for deletion, as dependent records have already been deleted
-      new_name = "DELETE-#{SecureRandom.uuid}"
-      object.assign_attributes({queued_for_deletion: true, upload_file_name: new_name, name: new_name, file_type: 'DELETE'})
-      object.save!(validate: false)
+      # it is safe to call :destroy now as all child documents have been removed
+      object.destroy
```

Comment:
oh, very cool, so we'll just no longer have the renamed files lingering around for cleanup, we can just delete them straightway.  What was the original reason we weren't deleting them here?

Reply:
The original reason was that calling `destroy` on a `StudyFile` would invoke all of the callbacks on child records, and since the `has_many` associations all have the `dependent: :destroy` clause, this could take quite a while. This was why we initially created `DeleteQueueJob` years ago.  

Over time we've gotten more proactive about cleanup up child records before deletion, and now by the time we get to this line, we've already removed all of the dependent documents, so calling `destroy` doesn't take any extra time.  This was probably something we could have changed a while ago but there wasn't any compelling reason to (until now).